Upgrade to ubunutu 20.04

FireballDWF opened the pull request:
- [ ] update ami's for china when they become available
- [ ] Put the CreationPolicy back
- [ ] optional support client-type=advanced
- [ ] change access to IMDS to use v2 Token method

FireballDWF committed:
add update

FireballDWF committed:
add update

FireballDWF committed:
add update

FireballDWF committed:
add update

FireballDWF committed:
-works on ubunutu 20.04 in us-east-1, ami-ids for other reasons not changed from 18.04 yet

FireballDWF committed:
-works on ubunutu 20.04

FireballDWF committed:
-works on ubunutu 20.04

FireballDWF committed:
added comment

FireballDWF committed:
added comment

FireballDWF committed:
added comment

jkataja commented:
Thanks for the PR! I am reluctant to update the template to Ubuntu 20.04, because there does not seem to be important new features or improvements. Also the current Ubuntu 18.04 has lifetime remaining until Apr 2023. The Folding@home dependency and NVidia CUDA drivers are better tested or built against Ubuntu 18.04.

Ubuntu 20.04 image ids from Ubuntu site, see: https://cloud-images.ubuntu.com/locator/ec2/

af-south-1 ami-033b88f80930a0f7a
ap-east-1 ami-9ccc8aed
ap-northeast-1 ami-0c1ac8728ef7f87a4
ap-northeast-2 ami-0de407404c33d1671
ap-northeast-3 ami-0e319adbbd61f8b6f
ap-south-1 ami-0c5b1a88222ac79cb
ap-southeast-1 ami-0b8cf0f359b1335e1
ap-southeast-2 ami-0a1a4d97d4af3009b
ca-central-1 ami-08a6203f59f9df866
eu-central-1 ami-05c26ae4789875080
eu-north-1 ami-0d4e2b57f569e9daa
eu-west-1 ami-0dad359ff462124ca
eu-west-2 ami-0917237b4e71c5759
eu-west-3 ami-0e60c6afa19d896ee
me-south-1 ami-09cbb7fde4a0edad0
sa-east-1 ami-001c16a3a4f5d85f1
us-east-1 ami-068663a3c619dd892
us-east-2 ami-0e84e211558a022c0
us-west-1 ami-075fd582acf0c0128
us-west-2 ami-09dd2e08d601bff67

FireballDWF commented:
Ok, I'll do the other changes I plan to do as independent PR's, so you can pick what you want to include in your repo.

FireballDWF committed:
restrict ingress rule by cidr

FireballDWF committed:
updated ami list

    pick 7ae79ab added comment
    pick 75ace9c updated ami list

FireballDWF committed:
updated ami list

FireballDWF committed:
updated ami list

    pick 7ae79ab added comment
    pick 75ace9c updated ami list

FireballDWF committed:
rebased

FireballDWF committed:
rebased

FireballDWF committed:
Merge branch 'upgrade-20.04' of github.com:FireballDWF/cfn-foldingathome into upgrade-20.04

jkataja committed:
remove apt-get upgrade

 - to be added later in separate branch, needs reboot

jkataja committed:
document SSHLocation parameter

jkataja committed:
sort order of parameters in README

jkataja committed:
version 20.4.26

jkataja committed:
attempt to supress Checkov failing

jkataja committed:
another try of Checkov supress

jkataja committed:
remove Checkov action

 - could not add skip comments
 - see https://github.com/bridgecrewio/checkov/issues/234

jkataja committed:
add cfn-lint-action

jkataja committed:
fix action parse error

jkataja committed:
remove wildcard

jkataja committed:
fix file suffix

jkataja committed:
add linter badge

FireballDWF committed:
SignalResource ready for test

FireballDWF committed:
fix merge conflict
